### PR TITLE
[Baremetal] Introduce DER encoded test certificates

### PR DESCRIPTION
**Pull request by hanno-arm**
This is the sideport of #2497 to the baremetal branch, introducing DER encoded test certificates.

This allows to remove `MBEDTLS_PEM_PARSE_C` and `MBEDTLS_BASE64_C` from the modification file `configs/baremetal-test.h` to `configs/baremetal.h` necessary to run `ssl_client2` and `ssl_server2`.

This PR is based on #2524.

### CODE REVIEW

**Comment by mpg** [comment]
(Note: changed the base twice in order to flush github's commit cache.)

**Comment by mpg** [comment]
... and removing "needs: preceding PR" as #2524 has been merged now.

**Comment by mpg** [comment]
Note: Jenkins hasn't been able to run due to github API rate limiting. I can't find a way to restart the jobs, so I'm applying "needs: CI" so that we remember, and in the meantime moving on to review - if I'm requesting changes then pushing those changes should unblock the CI.

**Comment by hanno-arm** [comment]
I re-committed the latest commit and force-pushed to retrigger the CI.

File: tests/suites/test_suite_x509parse.data [hunk 1]
```diff
@@ -2,13 +2,25 @@ X509 Certificate information #1
 depends_on:MBEDTLS_PEM_PARSE_C:MBEDTLS_RSA_C:MBEDTLS_SHA1_C
 x509_cert_info:"data_files/server1.crt":"cert. version     \: 3\nserial number     \: 01\nissuer name       \: C=NL, O=PolarSSL, CN=PolarSSL Test CA\nsubject name      \: C=NL, O=PolarSSL, CN=PolarSSL Server 1\nissued  on        \: 2011-02-12 14\:44\:06\nexpires on        \: 2021-02-12 14\:44\:06\nsigned using      \: RSA with SHA1\nRSA key size      \: 2048 bits\nbasic constraints \: CA=false\n"
 
+X509 Certificate information #1 (DER)
+depends_on:MBEDTLS_PEM_PARSE_C:MBEDTLS_RSA_C:MBEDTLS_SHA1_C
```
Comment (mpg) [inline tests/suites/test_suite_x509parse.data]: We probably want to remove `PEM_PARSE_C` from the dependencies here as well.

**Comment by mpg** [comment]
@sbutcher-arm @hanno-arm Since @jarlamsa is not available to review this soon, we need to find another reviewer. But I'm actually wondering: since I understand we'll be periodically merging from 2.16 to baremetal, and this already has a 2.16 backport, do we really need the baremetal sideport as well? Or should we rather focus on helping the 2.16 backport get merged?

**Comment by hanno-arm** [comment]
@mpg Good point - as this is not urgent, I think we can focus on the 2.16 backport and wait for that to get merged into here.

**Pull request closed without merging**